commit ab6c45f5249d077463c1876d7cf09a11f04240fa
Author: Manabu Matsui <manabu.matsui@gmail.com>
Date:   Wed Nov 22 11:07:15 2017 +0900

    Fix bug #75533: array_reduce is slow when $carry is large array

diff --git a/ext/standard/array.c b/ext/standard/array.c
index 4ba286e..15df3b9 100644
--- a/ext/standard/array.c
+++ b/ext/standard/array.c
@@ -5080,18 +5080,17 @@ PHP_FUNCTION(array_reduce)
 	fci.no_separation = 0;
 
 	ZEND_HASH_FOREACH_VAL(htbl, operand) {
-		ZVAL_COPY(&args[0], &result);
+		ZVAL_COPY_VALUE(&args[0], &result);
 		ZVAL_COPY(&args[1], operand);
 		fci.params = args;
 
 		if (zend_call_function(&fci, &fci_cache) == SUCCESS && Z_TYPE(retval) != IS_UNDEF) {
 			zval_ptr_dtor(&args[1]);
 			zval_ptr_dtor(&args[0]);
-			zval_ptr_dtor(&result);
 			ZVAL_COPY_VALUE(&result, &retval);
 		} else {
 			zval_ptr_dtor(&args[1]);
 			zval_ptr_dtor(&args[0]);
 			return;
 		}
 	} ZEND_HASH_FOREACH_END();
